File: frontend/src/app/api/returns.ts
import axios from 'axios';
import { useMutation, useQueryClient } from '@tanstack/react-query';

const API_BASE_URL = '/api/v1';

export interface ReturnPayload {
  lotId: number;
  quantity: number;
  reason: string;
  reference?: string;
}

export interface ReturnResult {
  returnId: number;
  returnNumber: string;
  returnType: string;
  lotId: number;
  quantity: number;
  status: string;
  requiresIqc: boolean;
}

export async function customerReturn(payload: ReturnPayload): Promise<ReturnResult> {
  const response = await axios.post(`${API_BASE_URL}/returns/customer`, payload);
  return response.data;
}

export async function supplierReturn(payload: ReturnPayload): Promise<ReturnResult> {
  const response = await axios.post(`${API_BASE_URL}/returns/supplier`, payload);
  return response.data;
}

export function useCustomerReturnMutation() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: customerReturn,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['inventory-lots'] });
    },
  });
}

export function useSupplierReturnMutation() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: supplierReturn,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['inventory-lots'] });
    },
  });
}

File: backend/app/schemas/returns.py
from pydantic import BaseModel, Field


class CustomerReturnCreate(BaseModel):
    lotId: int
    quantity: int = Field(gt=0)
    reason: str = Field(min_length=1, max_length=1000)
    reference: str | None = Field(default=None, max_length=100)


class SupplierReturnCreate(BaseModel):
    lotId: int
    quantity: int = Field(gt=0)
    reason: str = Field(min_length=1, max_length=1000)
    reference: str | None = Field(default=None, max_length=100)


class ReturnOut(BaseModel):
    returnId: int
    returnNumber: str
    returnType: str
    lotId: int
    quantity: int
    status: str
    requiresIqc: bool
